contracts/L2/gateway/L2LPTGateway.sol

```diff
@@ -43,6 +53,7 @@ contract L2LPTGateway is IL2LPTGateway, ControlledGateway, L2ArbitrumMessenger {
         require(extraData.length == 0, "CALL_HOOK_DATA_NOT_ALLOWED");
 
         Mintable(l2Lpt).burn(from, _amount);
+        IL2LPTDataCache(l2LPTDataCache).decreaseL2SupplyFromL1(_amount);
```

Comment:
In case of a mass withdraw back to L1 this will hit an underflow causing a revert as totalSupply on L2 will be > l2SupplyFromL1 accounting inflation. 

One solution could be to add a conditional and only decrease this value until it reaches 0. In that case it will be safe to assume that the remaining amount was created by inflation and wont affect the other calculations.

Reply:
Actually the above solution won't work as the "extra" minted tokens will increase the L1 total supply which then needs to be reported back to L2 to maintain correct L1 circulating amount.

Reply:
> Actually the above solution won't work as the "extra" minted tokens will increase the L1 total supply which then needs to be reported back to L2 to maintain correct L1 circulating amount.

That should be fine. In that scenario, the L1 total supply will be increased and that value should be cached in L2LPTDataCache via a call to L1LPTDataCache.cacheTotalSupply(). Then, in L2LPTDataCache, l2SupplyFromL1 = 0 because there is no LPT on L2 that is from L1. So, the L1 circulating supply on L2 is just L2LPTDataCache.totalSupply().

Reply:
> One solution could be to add a conditional and only decrease this value until it reaches 0. In that case it will be safe to assume that the remaining amount was created by inflation and wont affect the other calculations.

Implemented in 69e0754